_posts/2017-04-27-Rust-1.17.md

```diff
@@ -0,0 +1,352 @@
+```
+
+In some situations, this can remove lots of boilerplate:
+
+```rust
+// old
+const NAMES: &'static [&'static str; 2] = &["Ferris", "Bors"];
+
+// new
+const NAMES: &[&str; 2] = &["Ferris", "Bors"];
+```
+
+Another similar improvement is "field init shorthand." Similar to ECMAScript 6,
+which calls this "Object Literal Property Value Shorthand", duplication can be
+removed when declaring structs, like this:
+
+```rust
+// definitions
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+let x = 5;
+let y = 6;
+
+// old
+let p = Point {
+    x: x,
+    y: y,
+};
+
+// new
+let p = Point {
+    x,
+    y,
+};
+```
+
+That is, the `x, y` form will assume that its values are set to a varaible
+with the same name in its scope.
+
+In a similar way, it's common for new Rustaceans to try to use `+` to add two
+`&str`s together. This doesn't work, you can only add `String + &str`. As
+such, [a new error message](https://github.com/rust-lang/rust/pull/39116)
+was added to help users who make this mistake:
+
+```rust
+// code
+"foo" + "bar"
+
+// old
+error[E0369]: binary operation `+` cannot be applied to type `&'static str`
+ --> <anon>:2:5
+  |
+2 |     "foo" + "bar"
+  |     ^^^^^
+  |
+note: an implementation of `std::ops::Add` might be missing for `&'static str`
+ --> <anon>:2:5
+  |
+2 |     "foo" + "bar"
+  |     ^^^^^
+
+// new
+error[E0369]: binary operation `+` cannot be applied to type `&'static str`
+ --> <anon>:2:5
+  |
+2 |     "foo" + "bar"
+  |     ^^^^^
+  |
+  = note: `+` can't be used to concatenate two `&str` strings
+help: to_owned() can be used to create an owned `String` from a string
+reference. String concatenation appends the string on the right to the string on
+the left and may require reallocation. This requires ownership of the string on
+the left.
+  |     "foo".to_owned() + "bar"
+```
+
+When using Cargo's build scripts, you must set the location of the script in your
+`Cargo.toml`. However, the vast majority of people wrote `build = "build.rs"`, using
+a `build.rs` file in the root of their project. [This convention is now encoded
+into Cargo](https://github.com/rust-lang/cargo/pull/3664), and will be assumed if
+`build.rs` exists. We've been warning about this change for the past few releases,
+and you can use `build = false` to opt out.
+
+This release marks [the removal](https://github.com/rust-lang/rust/pull/39431)
+of the old `Makefile` based build system. We announced this new system in 1.15,
+which is written in Rust and primarily uses Cargo to drive the build, and now
+it is mature enough to always use.
+
+As part of that change, packages from crates.io can now be used within Rust's
+build system. The first one to be added was [mdBook](https://crates.io/crates/mdbook),
+and [it's now being used](https://github.com/rust-lang/rust/pull/39633) to render
+our various book-like documentation:
+
+* [The book](https://doc.rust-lang.org/stable/book/) ([repo](https://github.com/rust-lang/book))
```

Comment:
This isn't generally applicable to all /stable/ links, but it could be appropriate to link these to /1.17.0/ rather than /stable/, since the point of the links was to showcase some kind of specific difference in rendering.  If the rendering changes again in the future, the link to /stable/ would probably land the user on something that wasn't actually generated with mdBook.